x-pack/filebeat/input/httpinput/config.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpinput
+
+// Config contains information about httpjson configuration
+type config struct {
+	UseSSL          bool   `config:"ssl"`
```

Comment:
All of the TLS options can be defined by reusing `tlscommon`.


```suggestion
	TLS  *tlscommon.ServerConfig `config:"ssl"`
```

Reply:
That's a good point @andrewkroh . I looked at it briefly, as I saw some other inputs used it, and it was a bit unclear which of my options was covered by it and not. If I include tlscommon, does it only cover the settings in config.go located in that folder?
Ref:
```
type Config struct {
	Enabled          *bool                   `config:"enabled" yaml:"enabled,omitempty"`
	VerificationMode TLSVerificationMode     `config:"verification_mode" yaml:"verification_mode"` // one of 'none', 'full'
	Versions         []TLSVersion            `config:"supported_protocols" yaml:"supported_protocols,omitempty"`
	CipherSuites     []tlsCipherSuite        `config:"cipher_suites" yaml:"cipher_suites,omitempty"`
	CAs              []string                `config:"certificate_authorities" yaml:"certificate_authorities,omitempty"`
	Certificate      CertificateConfig       `config:",inline" yaml:",inline"`
	CurveTypes       []tlsCurveType          `config:"curve_types" yaml:"curve_types,omitempty"`
	Renegotiation    tlsRenegotiationSupport `config:"renegotiation" yaml:"renegotiation"`
	CASha256         []string                `config:"ca_sha256" yaml:"ca_sha256,omitempty"`
}
```

Reply:
Correct, that's what it covers. You need to provide all the other non-TLS options. By reusing it users get a consistent set of options across the stack.

https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/libbeat/common/transport/tlscommon/server_config.go#L28-L38

Reply:
Okay thanks will add it in for the next version then! Always appreciate feedback, will wait a bit for more comments if there is any and implement all of them in one go :)